test-basic/test-internal.js

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2023 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+let internal = require('../lib/internal.js');
+const testconfig = require("../etc/test-config");
+const marklogic = require("../lib/marklogic");
+const should = require('should');
+const dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnection);
+
+describe('internal tests', function() {
+
+    after((function(done){
+        marklogic.releaseClient(dbWriter);
+        done();
+    }));
+
+    it('test internal object exists', function(done){
+
+        try {
+            should.exist(dbWriter.internal);
+            done();
+        } catch(error){
+            done(error);
+        }
+    });
+
+    it('test newRequestOptions creation', function(done){
+        const requestOptions = dbWriter.internal.newRequestOptions('/v1/ping');
+        requestOptions.method = 'POST';
+        requestOptions.headers = {
+            'Accept':       'application/json',
+            'Content-Type': 'application/json'
+        };
+        try {
+            should.deepEqual(requestOptions.path, '/v1/ping');
```

Comment:
I think the main thing to test here is that the optional "database" and "basePath" params are accounted for. I don't think you need to test the values of "method" and "headers", as the user is setting those themselves. But the user would expect "path" to account for "database" and "basePath" so that the user doesn't need to account for them.

Reply:
Adding a new test for this.